sympy/polys/multivariate_resultants.py

```diff
@@ -0,0 +1,339 @@
+    [-a_1,  a_0,  a_2,    0],
+    [   0, -a_1,    0,    0],
+    [   0,    0, -a_1,    0],
+    [   0,    0,    0, -a_1]])
+
+    Reference
+    ==========
+    1. [Bruce97]_
+    2. [Stiller96]_
+
+    See Also
+    ========
+    Notebook in examples: sympy/example/notebooks.
+    """
+    def __init__(self, polynomials, variables):
+        """
+        Parameters
+        ----------
+        variables: list
+            A list of all n variables
+        polynomials : list of sympy polynomials
+            A  list of m n-degree polynomials
+        """
+        self.polynomials = polynomials
+        self.variables = variables
+        self.n = len(variables)
+
+        # A list of the d_max of each variable.
+        self.degrees = [total_degree(poly, *self.variables) for poly in self.polynomials]
+
+        self.degree_m = self._get_degree_m()
+        self.monomials_size = self.get_size()
+
+    def _get_degree_m(self):
+        r"""
+        Returns
+        -------
+        degree_m: int
+            The degree_m is calculated as  1 + \sum_1 ^ n (d_i - 1), where
+            d_i is the degree of the i polynomial
+        """
+        return 1 + sum(d - 1 for d in self.degrees)
+
+    def get_size(self):
+        r"""
+        Returns
+        -------
+        size: int
+            The size of set T. Set T is the set of all possible monomials of
+            the n variables for degree equal to the degree_m
+        """
+        return binomial(self.degree_m + self.n - 1, self.n - 1)
+
+    def get_monomials_of_certain_degree(self, degree):
+        """
+        Returns
+        -------
+        monomials: list
+            A list of monomials of a certain degree.
+        """
+        monomials = [Mul(*monomial) for monomial in combinations_with_replacement(self.variables, degree)]
+
+        return sorted(monomials, key=monomial_key('lex', self.variables), reverse=True)
+
+    def get_monomials_set(self):
+        r"""
+        Returns
+        -------
+        self.monomial_set: set
+            The set T. Set of all possible monomials of degree degree_m
+        """
+        monomial_set = self.get_monomials_of_certain_degree(self.degree_m)
+        self.monomial_set = monomial_set
+
+    def get_row_coefficients(self):
+        """
+        Returns
+        -------
+        row_coefficients: list
+            The row coefficients of Macaulay's matrix
+        """
+        row_coefficients = []
+        divisible = []
+        for i in range(self.n):
+            if i == 0:
+                row_coefficients.append(self.get_monomials_of_certain_degree(self.degree_m - self.degrees[i]))
+            else:
+                divisible.append(self.variables[i - 1] ** self.degrees[i - 1])
+                poss_rows = self.get_monomials_of_certain_degree(self.degree_m - self.degrees[i])
+                for div in divisible:
+                    for p in poss_rows:
+                        if fraction((p / div).expand())[1] == 1:
```

Comment:
To see that `p` is divisible by `div`, we can check the remainder: `rem(p, div) == 0`.